flink-rpc/flink-rpc-akka/src/test/java/org/apache/flink/runtime/rpc/akka/AkkaRpcServiceTest.java

```diff
@@ -85,15 +85,16 @@ static void shutdown() throws InterruptedException, ExecutionException, TimeoutE
     // ------------------------------------------------------------------------
     //  tests
     // ------------------------------------------------------------------------
-
     @Test
     void testScheduleRunnable() throws Exception {
```

Comment:
I'm wondering whether we should move this test into `ActorSystemScheduledExecutorAdapterTest` instead of having this being tested in `AkkaRpcServiceTest` :thinking:

Reply:
We could check whether `AkkaRpcService#getScheduledExecutor` returns a `ActorSystemScheduledExecutorAdapter` to be on the safe side. ...since that's some kind of implementation detail that we might want to ensure. :thinking:

Reply:
> We could check whether AkkaRpcService#getScheduledExecutor returns a ActorSystemScheduledExecutorAdapter to be on the safe side. ...since that's some kind of implementation detail that we might want to ensure. thinking

It doesn't matter what kind of executor class it returns, so long as it adheres to the contract of setting the context classloader. We'd just be testing implementation details.

Reply:
> I'm wondering whether we should move this test into ActorSystemScheduledExecutorAdapterTest instead of having this being tested in AkkaRpcServiceTest thinking

We'd need these tests anyway on the rpc service level (to ensure it actually uses something internally to ensure the context CL setting), so we'd just duplicate tests.

Reply:
Arguably we could test it in ActorSystemScheduledExecutorAdapterTest and assert that the rpc service returns a ActorSystemScheduledExecutorAdapter, but it feels strange to do that.